lab_3_generate_by_ngrams/main.py

```diff
@@ -93,6 +127,23 @@ def encode(self, text: str) -> Optional[tuple[int, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or len(text) == 0:
+            return None
+        encoding = {}
+        encoded_corpus = []
+        tokens = self._tokenize(text)
+        if tokens is None:
+            return None
+        for i in range(len(tokens)):
```

Comment:
Why not iterating over the values directly instead of indices?

Reply:
Done